src/main/java/redis/clients/util/Pool.java

```diff
@@ -48,6 +49,11 @@ public T getResource() {
     try {
       return internalPool.borrowObject();
     } catch (NoSuchElementException nse) {
+      if (null == nse.getCause()) { // The exception was caused by an exhausted pool
```

Comment:
I meant to add it to `Pool.java` where `GenericObjectPool<T> internalPool` is declared (and from which `JedisPool` inherits)